p2p/p2p.go

```diff
@@ -71,115 +100,165 @@ func (p2p *P2P) Run() {
 	go func() {
```

Comment:
Instead of doing query peers and stats check in the same goroutine, separate them into two different goroutines.
The problem here is, if the `queryPeer` request fails, the stats request will not be sent and their stats won't be updated.